lib/azure/armrest/model/storage_account.rb

```diff
@@ -119,6 +123,287 @@ def table_data(name, key = access_key, options = {})
         data
       end
 
+      ### Files and Directories
+
+      # Create a new directory under the specified +share+ or parent directory.
+      #
+      # The only supported option at this time is a "timeout" option.
+      #
+      def create_directory(share, directory, key = access_key, options = {})
+        raise ArgumentError, "No access key specified" unless key
+
+        query = {:restype => 'directory'}.merge(options).to_query
+
+        response = file_response(key, query, 'put', '', File.join(share, directory))
+
+        headers = Azure::Armrest::ResponseHeaders.new(response.headers)
```

Comment:
These three lines have appear so many times. We can make them into a shared method
```
def response_headers(response)
  Azure::Armrest::ResponseHeaders.new(response.headers).tap do |headers|
    headers.response_code = response.code
  end
end
private response_headers
```